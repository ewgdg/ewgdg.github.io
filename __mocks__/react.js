const React = jest.requireActual("react")
module.exports = {
  ...React,
  useContext: jest.fn().mockImplementation(React.useContext),
}
