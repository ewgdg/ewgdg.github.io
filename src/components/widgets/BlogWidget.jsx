import React from "react"
import StyledTitle from "components/titles/StyledTitle"
import CardContainer from "components/thumbnail/CardContainer"
import MediaCard from "components/thumbnail/MediaCard"
import Container from "@material-ui/core/Container"
import Grid from "@material-ui/core/Grid"
import Button from "@material-ui/core/Button"
import FlexContainer from "../sections/FlexContainer"

function BlogPreview() {
  return (
    <FlexContainer>
      <Container style={{ maxHeight: "100%", height: "600px" }}>
        <div style={{ height: "15%" }}>
          <StyledTitle title="Blog" style={{ height: "50%" }} />
          <p
            style={{
              display: "block",
              textAlign: "center",
              height: "50%",
              boxSizing: "border-box",
            }}
          >
            Know what I am thinking:
          </p>
        </div>
        <div style={{ height: "75%" }}>
          <CardContainer style={{ marginBottom: "25px" }}>
            <MediaCard />
            <MediaCard />
          </CardContainer>
        </div>
        <Grid container justify="center" style={{ maxHeight: "10%" }}>
          <Button variant="outlined" color="secondary">
            View more
          </Button>
        </Grid>
      </Container>
    </FlexContainer>
  )
}

export default BlogPreview
