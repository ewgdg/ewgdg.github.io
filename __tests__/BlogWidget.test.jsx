import React from "react"
import { render } from "@testing-library/react"
import { useStaticQuery } from "gatsby"
import BlogWidget from "../src/components/widgets/BlogWidget"

describe("BlogWidget", () => {
  beforeAll(() => {
    // fake post data
    useStaticQuery.mockReturnValue({
      allMarkdownRemark: {
        edges: [
          {
            post: {
              excerpt: "excerpt",
              id: "id",
              fields: {
                slug: "slug",
              },
              frontmatter: {
                title: "title",
                description: "description",
                templateKey: "templateKey",
                externalLink: null,
                date: "SEPT 09, 1999",
                tags: "tags",
                featuredPost: false,
                featuredImage: {
                  childImageSharp: {
                    fluid: {
                      src: "/images/icon.png",
                    },
                  },
                },
              },
            },
          },
        ],
      },
    })
  })
  it("should render properly", () => {
    const { container } = render(<BlogWidget />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
