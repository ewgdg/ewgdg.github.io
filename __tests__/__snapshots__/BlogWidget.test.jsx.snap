// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogWidget should render properly 1`] = `
<div
  class="makeStyles-flexContainer-1 undefined"
>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
    style="max-height: 100%; height: 600px;"
  >
    <div
      style="height: 15%;"
    >
      <div
        style="text-align: center; height: 50%;"
      >
        <h2
          class="makeStyles-heading-9"
        >
          <strong>
             
            Blog
             
          </strong>
        </h2>
      </div>
      <p
        style="display: block; text-align: center; height: 50%; box-sizing: border-box;"
      >
        Know what I am thinking:
      </p>
    </div>
    <div
      style="height: 75%;"
    />
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-flex-end MuiGrid-justify-xs-center"
      style="max-height: 10%;"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          View more
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
